airbyte-integrations/connectors/source-db2/src/main/java/io.airbyte.integrations.source.db2/Db2SourceOperations.java

```diff
@@ -69,4 +75,31 @@ private void putDecfloat(final ObjectNode node,
     }
   }
 
+  @Override
+  protected void putDate(final ObjectNode node,
+                         final String columnName,
+                         final ResultSet resultSet,
+                         final int index)
+          throws SQLException {
+    final Date date = resultSet.getDate(index);
+    node.put(columnName, DateTimeConverter.convertToDate(date));
+  }
+
+  @Override
+  protected void putTime(final ObjectNode node,
+                         final String columnName,
+                         final ResultSet resultSet,
+                         final int index)
+          throws SQLException {
+    // resultSet.getTime() will lose nanoseconds precision
+    final LocalTime localTime = resultSet.getTimestamp(index).toLocalDateTime().toLocalTime();
+    node.put(columnName, DateTimeConverter.convertToTime(localTime));
```

Comment:
it looks like this code is identical to redshift / snowflake? Assuming I didn't just miss the difference - does it make sense to extract it to a shared method (maybe in DateTimeConverter?)

Reply:
> it looks like this code is identical to redshift / snowflake? Assuming I didn't just miss the difference - does it make sense to extract it to a shared method (maybe in DateTimeConverter?)

extracted to the separate methods _putJavaSQLDate_() and _putJavaSQLTime_()